models/base.py

```diff
@@ -424,18 +437,27 @@ def get_name_in_current_ctx(self) -> str:
     """
     Get layer name valid for current scope.
     """
+    """
+    Get layer name valid for current scope.
+  """
     cur_scope = NameCtx.current_ctx()
     if self.parent is cur_scope:  # fast path
       return self.name
     cur_scope_abs = cur_scope.get_abs_name_ctx_list()
     self_name_abs = self.get_abs_name_ctx_list()
     assert cur_scope_abs[0] is self_name_abs[0]  # same root
-    assert len(cur_scope_abs) >= len(self_name_abs)  # not implemented otherwise
     common_len = 0
-    while cur_scope_abs[common_len + 1] is self_name_abs[common_len + 1]:
+    max_common_len = min(len(cur_scope_abs), len(self_name_abs))
+    while common_len + 1 < max_common_len and cur_scope_abs[common_len + 1] is self_name_abs[common_len + 1]:
       common_len += 1
-    assert common_len == len(self_name_abs) - 2  # not implemented otherwise
-    return "base:" * (len(cur_scope_abs) - len(self_name_abs) + 1) + self.name
+    # assert common_len == len(self_name_abs) - 2  # not implemented otherwise
+    full_name = self_name_abs[-1].get_abs_name()
+    subs = full_name.split("/")
+    relative_name = subs[len(cur_scope_abs)-1:-1]
+    prefix = "/".join(relative_name)
+    if not prefix == "":
+      prefix += "/"
+    return "base:" * (len(cur_scope_abs) - len(self_name_abs) + 1) + prefix + self.name
```

Comment:
I'm pretty sure you miss `common_len` here. See the `relpath` logic. This would correspond to `commonprefix`.